packages/origin-ui-core/src/components/Organization/OrganizationView.tsx

```diff
@@ -43,22 +46,38 @@ export function OrganizationView() {
 
     const classes = useStyles(useTheme());
 
+    const setValues = (organization, activeCountriesParsed) => {
+        setFormValues({
+            ...organization,
+            headquartersCountry: Countries.find((c) => c.id === organization.headquartersCountry)
+                .name,
+            activeCountries: Countries.filter((c) => activeCountriesParsed.includes(c.id))
+                .map((country) => country.name)
+                .join(', '),
+            country: Countries.find((c) => c.id === organization.country).name
+        });
+    };
+
     useEffect(() => {
-        const organization = userOffchain?.organization;
-        if (organization) {
-            const activeCountriesParsed: number[] = JSON.parse(organization.activeCountries);
-            setFormValues({
-                ...organization,
-                headquartersCountry: Countries.find(
-                    (c) => c.id === organization.headquartersCountry
-                ).name,
-                activeCountries: Countries.filter((c) => activeCountriesParsed.includes(c.id))
-                    .map((country) => country.name)
-                    .join(', '),
-                country: Countries.find((c) => c.id === organization.country).name
-            });
+        if (!params.id) {
+            const organization = userOffchain?.organization;
+            if (organization) {
+                const activeCountriesParsed: number[] = JSON.parse(organization.activeCountries);
+                setValues(organization, activeCountriesParsed);
+            }
+        } else {
+            const getOrganization = async () => {
+                const organization = await organizationClient.getById(parseInt(params.id));
+                if (organization) {
+                    const activeCountriesParsed: number[] = JSON.parse(
+                        organization.activeCountries
+                    );
+                    setValues(organization, activeCountriesParsed);
+                }
+            };
+            getOrganization();
```

Comment:
Looks good from the logic perspective, but isn't this a one-line fix?

Changing 
```
const organization = userOffchain?.organization;
```
to:
```
const organization = params.id ? await organizationClient.getById(parseInt(params.id)) : userOffchain?.organization;
```
would avoid code duplication in `useEffect()`.